deltacat/exceptions.py

```diff
@@ -1,14 +1,145 @@
+    error_code = DeltaCatErrorMapping.DependencyRayError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A dependency Ray error occurred, during ray_task: {ray_task}."
+
+
+class DependencyRayWorkerDiedError(DependencyRayError):
+    error_code = DeltaCatErrorMapping.DependencyRayWorkerDiedError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. Ray Worker Died Unexpectedly, during ray_task: {ray_task}."
+
+
+class GeneralValidationError(DeltaCatError):
+    error_code = DeltaCatErrorMapping.GeneralValidationError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A validation error occurred."
+
+
+class ContentTypeValidationError(GeneralValidationError):
+    error_code = DeltaCatErrorMapping.ContentTypeValidationError.value
+    is_retryable = False
+    msg = (
+        "Error Code: {error_code}. Is Retryable Error: {is_retryable}. "
+        "S3 file with content type: {content_type} and content encoding: {content_encoding} "
+        "cannot be read into pyarrow.parquet.ParquetFile"
+    )
+
+
+class GeneralStorageError(DeltaCatError):
+    error_code = DeltaCatErrorMapping.GeneralStorageError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A general storage error occurred."
+
+
+class StorageConcurrentModificationError(GeneralStorageError):
+    error_code = DeltaCatErrorMapping.StorageConcurrentModificationError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A storage concurrent modification error occurred."
+
+
+class DependencyBotocoreError(DeltaCatError):
+    error_code = DeltaCatErrorMapping.DependencyBotocoreError.value
+    is_retryable = False
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A dependency botocore error occurred."
+
+
+class DownloadTableError(DependencyBotocoreError):
+    error_code = DeltaCatErrorMapping.DownloadTableError.value
+    is_retryable = False
+    msg = (
+        "Error Code: {error_code}. Is Retryable Error: {is_retryable}. "
+        "Botocore download table error occurred when downloading from {s3_url}."
+    )
+
+
+class UploadTableError(DependencyBotocoreError):
+    error_code = DeltaCatErrorMapping.UploadTableError.value
+    is_retryable = False
+    msg = (
+        "Error Code: {error_code}. Is Retryable Error: {is_retryable}. "
+        "Botocore upload table error occurred when uploading to {s3_url}."
+    )
+
+
+class GeneralThrottlingError(DeltaCatError):
+    error_code = DeltaCatErrorMapping.GeneralThrottlingError.value
+    is_retryable = True
+    msg = "Error Code: {error_code}. Is Retryable Error: {is_retryable}. A general throttling error occurred. Retried: {retry_attempts} times."
+
+
+class UploadTableThrottlingError(GeneralThrottlingError):
```

Comment:
We could wrap hb and merge task with an exception catcher and depending on the message string identified in oncall handoff we need to convert it to appropriate exception there.

Reply:
Yeah, will follow up with exception catcher and handler in second PR.

Reply:
Please add a TODO.